client/src/main/java/org/asynchttpclient/netty/handler/StreamedResponsePublisher.java

```diff
@@ -42,16 +42,18 @@ public StreamedResponsePublisher(EventExecutor executor, ChannelManager channelM
     protected void cancelled() {
         logger.debug("Subscriber cancelled, ignoring the rest of the body");
 
-        // The subscriber cancelled early, we need to drain the remaining elements from the stream
-        channelManager.drainChannelAndOffer(channel, future);
-        channel.pipeline().remove(StreamedResponsePublisher.class);
-
         try {
             future.done();
         } catch (Exception t) {
             // Never propagate exception once we know we are done.
-            logger.debug(t.getMessage(), t);
+            logger.error(t.getMessage(), t);
         }
+
+        // The subscriber cancelled early - this channel is dead and should be closed.
+        channelManager.releaseChannelLock(future.getPartitionKey());
```

Comment:
Releasing the lock is already done on future.done() above. You're most likely double releasing here.

Reply:
Will remove.